index.js

```diff
@@ -42,34 +48,33 @@ app.post('/', upload.single('thumb'), function (req, res, next) {
     const pauseEmoji = req.query.pauseEmoji || process.env.PAUSE_EMOJI || '';
     const statusEmoji = isPlayEvent ? playEmoji : pauseEmoji;
 
-    const params = [
-      ['token', token],
-      ['profile', JSON.stringify({
+    // If the user has set protected mode, do not change their status if they
+    // have a status emoji set that is neither the play or pause emoji.
+    const protected = req.query.protected === 'true';
+    if (protected) {
+      const { profile } = await slack.users.profile.get({ token });
```

Comment:
Maybe try/catch this so if it fails we go ahead and change the emoji as normal?

Reply:
If the user has explicitly set the `protected` option, I don't feel like updating their status if we can't check is the wrong behavior?

Also the most likely reason for it to fail is an invalid token, which would cause the `set` to fail as well.